src/test/java/com/emc/object/s3/S3JerseyClientTest.java

```diff
@@ -199,6 +210,228 @@ public void testCreateEncryptedBucket() {
         client.deleteBucket(bucketName);
     }
 
+    @Test
+    public void testEnableObjectLockOnExistingBucket() {
+        Assume.assumeTrue("Skip Object Lock related tests for non IAM user.", testIAM);
+
+        String bucketName = getTestBucket();
+        ObjectLockConfiguration objectLockConfig = client.getObjectLockConfiguration(bucketName);
+        Assert.assertNull(objectLockConfig);
+        client.enableObjectLock(bucketName);
+        objectLockConfig = client.getObjectLockConfiguration(bucketName);
+        Assert.assertEquals(ObjectLockConfiguration.ObjectLockEnabled.Enabled.toString(), objectLockConfig.getObjectLockEnabled());
+    }
+
+    @Test
+    public void testCreateObjectLockBucket() {
+        Assume.assumeTrue("Skip Object Lock related tests for non IAM user.", testIAM);
+
+        String bucketName = "s3-client-test-createObjectLockBucket";
+        client.createBucket(new CreateBucketRequest(bucketName).withObjectLockEnabledForBucket(true));
+        ObjectLockConfiguration objectLockConfig = client.getObjectLockConfiguration(bucketName);
+        Assert.assertEquals("Enabled", objectLockConfig.getObjectLockEnabled());
+        client.deleteBucket(bucketName);
+    }
+
+    @Test
+    public void testSetObjectLockConfiguration() {
+        Assume.assumeTrue("Skip Object Lock related tests for non IAM user.", testIAM);
+
+        String bucketName = getTestBucket();
+        ObjectLockConfiguration objectLockConfig = new ObjectLockConfiguration().withObjectLockEnabled(ObjectLockConfiguration.ObjectLockEnabled.Enabled);
+        DefaultRetention defaultRetention = new DefaultRetention().withMode(ObjectLockRetentionMode.GOVERNANCE).withDays(2);
+        objectLockConfig.setRule(new ObjectLockRule().withDefaultRetention(defaultRetention));
+        try {
+            client.setObjectLockConfiguration(bucketName, objectLockConfig);
```

Comment:
add `Assert.fail()` here - if no exception is thrown, that should fail the test, correct?

Reply:
Great suggestion! fixed.